### PR TITLE
fix: Replace runtime enum `BotKind` with plain object

### PR DESCRIPTION
**Pull request by dj-stormtrooper**
With new Sergey's idea BotD API remains unchanged, only runtime under-the-hood behaviour is different

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -53,5 +53,6 @@ export {
   ProcessPayload,
   DistinctivePropertiesPayload,
   BotDetectionResult,
-  BotKind,
+  BotKindType as BotKind,
+  BotKind as BotKindEnum,
```
Comment (Finesse) [inline src/index.ts:57]: Having the object in the `BotKind` export is a more compatible solution (at least the JS user code won't break). I think it's ok to break the TypeScript compatibility a bit.

Comment (Finesse) [reply src/index.ts:57]: It turns out removing the enum can be done without breaking changes:

```ts
export const BotKind = {
  Awesomium: 'awesomium',
  Cef: 'cef',
  // ...
} as const;
export type BotKind = typeof Fruit[keyof typeof BotKind];
```

I've checked that importing and re-exporting such a combined identifier works.

File: src/types.ts [hunk 2]
```diff
@@ -31,27 +31,29 @@ export const enum State {
...
-  Selenium = 'selenium',
-  Sequentum = 'sequentum',
-  SlimerJS = 'slimerjs',
-  WebDriverIO = 'webdriverio',
-
-  WebDriver = 'webdriver',
-  HeadlessChrome = 'headless_chrome',
-  Unknown = 'unknown',
-}
+export const BotKind = {
```
Comment (Finesse) [inline src/types.ts:34]: I'd also add a comment telling why an object is used instead of an enum

File: src/types.ts [hunk 3]
```diff
@@ -31,27 +31,31 @@ export const enum State {
...
+  // Object is used instead of Typescript enum to avoid emitting IIFE which might be affected by further tree-shaking.
```
Comment (dj-stormtrooper) [inline src/types.ts:35]: Put it here because it's maintainer-faced info, don't think we need it in the JSDoc annotation

**Review by xnerhu — APPROVED** [review]

**Review by xnerhu — APPROVED** [review]

**Pull request merged**